com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/modelinference/ExtendableInferredModelAssociator.java

```diff
@@ -10,34 +10,85 @@
  *******************************************************************************/
 package com.avaloq.tools.ddk.xtext.modelinference;
 
+import java.util.ArrayDeque;
+import java.util.Deque;
+
 import org.apache.log4j.Logger;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
 import org.eclipse.xtext.resource.DerivedStateAwareResource;
 
+import com.avaloq.tools.ddk.xtext.scoping.ImplicitReferencesAdapter;
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 
 
 /**
  * Manager for associations between source and inferred model elements that allows for additional inference to be done by external providers (i.e. outside of a
  * language core plugin).
+ * Also this associator registers implicit inference dependencies.
+ * This is needed for those cases when a resource loads others to perform inference.
+ * Registering dependencies will ensure that the inference will be re-triggered once one (or many) of those 'other' resources are changed.
  */
 @Singleton
 public class ExtendableInferredModelAssociator extends InferredModelAssociator {

```

Comment:
The change is good, but I think it should to to the parent of this class - InferredModelAssociator. This class is meant to allow external contributions of inference logic. So make installedDerivedState final in the InferredModelAssociator and factor our the execution of inference into some new methods which everyone has to override. i.e. inferTargetModel.

Reply:
done